### ISSUE TITLE
Eradicate (ERA001) is triggered on well formated TODO

### ISSUE THREAD

**Issue by Cjkjvfnby** [astral-sh/ruff#7031]
`ruff 0.0.286`

When my TODO is bad (according to the TD rules), it's not reported as unused code
But when I fix TD warnings the code becomes a subject of ERA

Broken TODO
```python
def foo():
    # TODO: implement
    pass
```

```shell
ruff "foo.py"  --select TD002,ERA
TD002 Missing author in TODO; try: `# TODO(<author_name>): ...` or `# TODO @<author_name>: ...`
```

When I fix this warning
```python
def foo():
    # TODO(CJ): implement
    pass

```

```shell
ruff foo.py --select TD002,ERA
ERA001 [*] Found commented-out code
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

I expect that the TODO comment should not be marked as unused code.  And TD requirements were aligned with other rules.

**Comment by zanieb** [comment]
I think it makes sense to add an exception to ERA001 if feasible — the pattern used for matching violations is brittle and has a lot of false positives.

**Comment by charliermarsh** [comment]
I'm not sure why it triggers in one case but not the other, but yeah I definitely would've expected TODO to be in the allowlist already. Oops!
